### PR TITLE
Migrate to neutrino v9

### PR DESCRIPTION
**Pull request by helfi92**
Fixes Bug 1489273

Blocker: https://github.com/mozilla-frontend-infra/neutrino-preset-mozilla-frontend-infra/pull/20

### CODE REVIEW

File: .neutrinorc.js [hunk 1]
```diff
@@ -4,18 +4,18 @@ module.exports = {
       hot: false,
       babel: {
         plugins: [
-          require.resolve('babel-plugin-transform-object-rest-spread'),
+          require.resolve('@babel/plugin-proposal-object-rest-spread'),
```
Comment (edmorley) [inline .neutrinorc.js]: `plugin-proposal-object-rest-spread` is now included in `@babel/preset-env`, so should work out of the box :-)

File: package.json [hunk 2]
```diff
@@ -6,17 +6,22 @@
...
-    "test": "neutrino test"
+    "build": "webpack --mode production",
+    "start": "webpack-dev-server --mode development",
+    "lint": "eslint --cache --format codeframe --ext mjs,jsx,js src test",
+    "test": "mocha --require mocha.config.js --recursive"
   },
   "devDependencies": {
-    "@neutrinojs/mocha": "^8.3.0",
-    "neutrino": "^8.2.1",
+    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
```
Comment (edmorley) [inline package.json]: This can be removed along with the above

File: .neutrinorc.js [hunk 3]
```diff
@@ -1,26 +1,69 @@
...
+      // Tap into the existing Babel options and merge our
+      // decorator options *before* the rest of the existing
+      // Babel options
+      neutrino.config.module
+        .rule('compile')
+        .use('babel')
+        .tap(options =>
+          merge(
+            {
+              plugins: [
```
Comment (eliperelman) [inline .neutrinorc.js]: I think you should be able to specify these to the node preset:

```js
['@neutrinojs/node', {
  babel: {
    plugins: [
      [require.resolve('@babel/plugin-proposal-decorators'), { legacy: true }],
      require.resolve('@babel/plugin-proposal-class-properties'),
    ]
  }
}]
```

**Comment by helfi92** [comment]
Many thanks for reviewing the code @edmorley :)

**Comment by edmorley** [comment]
You're welcome! I notice there's 4000 lines removed in yarn.lock, even with the new integrity field being added. That's a lot fewer unused dependencies by skipping neutrino-preset-mozilla-frontend-infra! :-)

**Pull request merged**